apiserver/utility/utility.go

```diff
@@ -27,61 +26,31 @@ func ValidateBitriseConfigAndSecret(bitriseConfig, secretsConfig string) (*Warni
 	bitriseConfigBase64 := base64.StdEncoding.EncodeToString([]byte(bitriseConfig))
 	secretsConfigBase64 := base64.StdEncoding.EncodeToString([]byte(secretsConfig))
 
-	validateCmd := command.New("bitrise",
-		"-l=panic", "validate", "--format=json",
-		"--config-base64", bitriseConfigBase64,
-		"--inventory-base64", secretsConfigBase64)
-
-	combinedOut, cmdErr := validateCmd.RunAndReturnTrimmedCombinedOutput()
-
-	type BitriseCLIValidateItemModel struct {
-		IsValid  bool     `json:"is_valid"`
-		Error    string   `json:"error"`
-		Warnings []string `json:"warnings"`
-	}
-	type BitriseCLIValidateDataModel struct {
-		Config  BitriseCLIValidateItemModel `json:"config"`
-		Secrets BitriseCLIValidateItemModel `json:"secrets"`
-	}
-	type BitriseCLIValidateOutputModel struct {
-		Data  BitriseCLIValidateDataModel `json:"data"`
-		Error string                      `json:"error"`
-	}
-
-	var validationOutput BitriseCLIValidateOutputModel
-	if outputParseErr := json.NewDecoder(strings.NewReader(combinedOut)).Decode(&validationOutput); outputParseErr != nil {
-		if cmdErr != nil {
-			return nil, fmt.Errorf("Failed to run bitrise validate command, error: %s | 'bitrise validate' command output: %s", cmdErr, combinedOut)
-		}
-		return nil, fmt.Errorf("Failed to parse bitrise validate output, error: %s | 'bitrise validate' command output: %s", outputParseErr, combinedOut)
-	}
+	_, bitriseWarns, bitriseErr := cli.CreateBitriseConfigFromCLIParams(bitriseConfigBase64, "")
+	_, secretsErr := cli.CreateInventoryFromCLIParams(secretsConfigBase64, "")
 
 	errorStrs := []string{}
-	if !validationOutput.Data.Config.IsValid {
-		errorStrs = append(errorStrs, "Config validation error: "+validationOutput.Data.Config.Error)
+	if bitriseErr != nil {
+		errorStrs = append(errorStrs, "Config validation error: "+bitriseErr.Error())
 	}
-	if !validationOutput.Data.Secrets.IsValid {
-		errorStrs = append(errorStrs, "Secret validation error: "+validationOutput.Data.Secrets.Error)
+	if secretsErr != nil {
+		errorStrs = append(errorStrs, "Secret validation error: "+secretsErr.Error())
+	} else {
+		if secretsConfigBase64 == "" {
+			errorStrs = append(errorStrs, "Secret validation error: Validation failed: Empty secrets configuration")
```

Comment:
In the PR comments you state: `Other than that, the Bitrise CLI will not issue an error if an empty secrets file is given. In such cases, the output will simply omit the secrets key.`

Than why do we raise this error?

Reply:
Sorry, the phrasing wasn't clear. 

Although the CLI does not raise an error, because the output does not contain key in the validation result JSON, `validationOutput.Data.Secrets.IsValid` will be `false` and so the Workflow Editor will eventually treat this as an error case.

Related test case: https://github.com/bitrise-io/bitrise-workflow-editor/blob/master/apiserver/utility/utility_test.go#L117